app/models/data_center.rb

```diff
@@ -72,19 +72,25 @@ class DataCenter < ApplicationRecord
   has_many :validator_ips_active, through: :data_center_hosts
   has_many :validators, through: :data_center_hosts
   has_many :validator_score_v1s, through: :data_center_hosts
+  has_many :gossip_nodes, through: :data_center_hosts
 
   scope :for_api, -> { select(FIELDS_FOR_API) }
 
   scope :by_data_center_key, ->(data_center_keys) do
     where(data_center_key: data_center_keys)
   end
 
-  def to_builder
+  def to_builder(map_data: false, network: "mainnet")
     Jbuilder.new do |data_center|
       data_center.data_center_key self.data_center_key
       data_center.autonomous_system_number self.traits_autonomous_system_number
       data_center.latitude self.location_latitude
       data_center.longitude self.location_longitude
+      if map_data
+        data_center.country_name self.country_name
+        data_center.nodes_count self.gossip_nodes.where(network: network).count
+        data_center.validators_count self.validators.where(network: network).count
```

Comment:
Smells like n+1 since I can't see any preloading in the controller (same for the line above). I'd also use .size instead of .count.